File: 0011-container-with-most-water/0011-container-with-most-water.java
import java.util.Collections;

class Solution {
    public int maxArea(int[] height) {

        int right = height.length - 1; 
        int left = 0 ;

        int area = 0;

        while(true){
            int h = Math.min(height[left], height[right]); 
            int w = Math.abs(left - right); 
            
            area = Math.max(area, h*w);
            
            if(height[left] <= height[right]){
                left ++ ;
            }else if(height[left] > height[right]){
                right -- ;
            }

            if (left >= right){
                break; 
            }

        }

        return area ;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start, end = 0, len(height)-1
    
        answer = 0    
        while(start < end):
            
            # print(height[start],height[end], " ", start,end)
            
            answer = max(answer, min(height[start],height[end]) * (end-start))

            if(height[start] >=  height[end]):
                end -= 1 
            elif(height[start] <  height[end]):
                start += 1

        return answer